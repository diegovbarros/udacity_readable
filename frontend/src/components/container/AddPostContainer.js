import React,{PureComponent} from 'react'
import {withRouter } from 'react-router-dom'
import { connect } from 'react-redux'

import  AddPost  from '../presentational/AddPost'

import {Creators as CategoryCreators} from '../../store/features/category'
import {Creators as PostCreators} from '../../store/features/post'

/**
* @description 
* Componente que representa o container do formulário para inserção do Post
*/
class AddPostContainer extends PureComponent {

    componentDidMount() {
        this.props.getAllCategories()
    }

     /**
    * @description 
    * Faz o bind do objeto comentário para realizar a inserção ou atualização
    * @param {Event} post  Post
    */
    onHandlePost = (post) =>{
        const {addPost,updatePost} = this.props 
        console.log(post)
        if(post.is_new)
            addPost(post);
        else
            updatePost(post);

        this.setState({post : {}})
    
    }

    render() {
        const {categories,user} = this.props
        return (
            <AddPost
                categories={categories}
                user={user}
                onHandlePost={this.onHandlePost}
            />
        )
   }
}

function mapStateToProps (state) {
    const {categories,user} = state;
    return {
        categories : categories.map((cat) => ({
            'value' : cat.name,
            'label' : cat.name.toUpperCase()
        })),
        user
    }
}

function mapDispatchToProps (dispatch) {
    return {
       getAllCategories: ()   =>  dispatch(CategoryCreators.fetch()),
       addPost: (post) => {
            dispatch(PostCreators.add(post))
        },
        updatePost: (post) => {
            dispatch(PostCreators.update(post))
        }
    }
}


export default withRouter(connect(mapStateToProps,mapDispatchToProps)(AddPostContainer))
