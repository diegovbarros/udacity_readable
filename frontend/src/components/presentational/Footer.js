import React from 'react'
import PropTypes from 'prop-types';


const propTypes = {
    project : PropTypes.string.isRequired
};

const defaultProps = {
    project: 'Udacity - Leitura'
};

/**
* @description 
* Componente que representa o footer genérico da aplicação
*
* @constructor
* @param {String} project   Nome do projeto
*/
function Footer({project}) {
    return (
        <div className="sub-footer-copyright">
        <span>
            {project}
        </span>
    </div>
    
    )
}


Footer.propTypes    = propTypes;
Footer.defaultProps = defaultProps;

export default Footer
