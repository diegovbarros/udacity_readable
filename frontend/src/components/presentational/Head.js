import React from 'react'
import PropTypes from 'prop-types';


const propTypes = {
    project             : PropTypes.string.isRequired,
    description         : PropTypes.string.isRequired
};

const defaultProps = {
    project: 'Udacity - Leitura',
    description : 'Bem vindos ao Udacity Leitura, onde podemos discutir assuntos variados com uma gama enorme de pessoas'
};

/**
* @description 
* Componente o header genérico da aplicação
*
* @constructor
* @param {String} project       Nome do projeto
* @param {String} description   Breve descrição da página
*/
function Head({project,description}) {
    return (
        <div className="main-header">
            <div className="content-bg-wrap bg-group"></div>
            <div className="container">
                <div className="row">
                    <div className="col col-lg-8 m-auto col-md-8 col-sm-12 col-12">
                        <div className="main-header-content">
                            <h1>{project}</h1>
                            <p>{description}</p>
                        </div>
                    </div>
                </div>
            </div>
            <img className="img-bottom" src="/img/blog_bottom.png" alt="Blog"/>
        </div>
    )
}

Head.propTypes      = propTypes;
Head.defaultProps   = defaultProps;

export default Head
