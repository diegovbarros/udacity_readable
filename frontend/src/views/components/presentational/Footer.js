import React from "react"
import PropTypes from 'prop-types';


const propTypes = {
    description         : PropTypes.string.isRequired
};

/**
* @description 
* Componente o footer genérico da aplicação
*
* @constructor
* @param {String} description   Breve descrição para o footer
*/
function Footer({description}) {
    return (
        <div class="sub-footer-copyright">
        <span>
            {description}
        </span>
    </div>
    )
}

Footer.propTypes = propTypes;

export default Footer
