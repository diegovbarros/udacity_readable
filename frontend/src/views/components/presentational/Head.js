import React from "react"
import PropTypes from 'prop-types';


const propTypes = {
    title               : PropTypes.string.isRequired,
    description         : PropTypes.string.isRequired
};

/**
* @description 
* Componente o header genérico da aplicação
*
* @constructor
* @param {String} title         Título do Header
* @param {String} description   Breve descrição da página
*/
function Head({title,description}) {
    return (
        <div class="main-header">
            <div class="content-bg-wrap bg-group"></div>
            <div class="container">
                <div class="row">
                    <div class="col col-lg-8 m-auto col-md-8 col-sm-12 col-12">
                        <div class="main-header-content">
                            <h1>{title}</h1>
                            <p>{description}</p>
                        </div>
                    </div>
                </div>
            </div>
            <img class="img-bottom" src="img/blog_bottom.png" alt="Blog"/>
        </div>
    )
}

Head.propTypes = propTypes;

export default Head
