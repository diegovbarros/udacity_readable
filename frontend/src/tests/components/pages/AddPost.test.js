import React from 'react';
import ReactRouterEnzymeContext from 'react-router-enzyme-context';

import ConnectedAddPost,{AddPost} from '../../../components/pages/AddPost';
import Head from '../../../components/presentational/Head';
import AddPostContainer from '../../../components/container/AddPostContainer';
import Footer from '../../../components/presentational/Footer';


describe('[Page Component] AddPost', () => {
	const context 			= new ReactRouterEnzymeContext();
	let props 					= {}

	it('Shallow renders correctly', () => {
    expect(global.shallow(<AddPost {...props} />));
	});

	it('Shallow renders correctly', () => {
    expect(global.shallow(<ConnectedAddPost {...props} />));
	});

	it('Renders correctly', () => {
    expect(global.shallow(<ConnectedAddPost {...props} />)).toMatchSnapshot();
	});

	it('Shallow renders all components correctly', () => {
		const wrapper = global.shallow(<AddPost {...props} />,context.get());
		expect(wrapper.find(Head)).toHaveLength(1);
    expect(wrapper.find(AddPostContainer)).toHaveLength(1);
		expect(wrapper.find(Footer)).toHaveLength(1);
	});

});
