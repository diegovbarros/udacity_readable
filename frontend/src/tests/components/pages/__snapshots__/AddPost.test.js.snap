// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Page Component] AddPost Renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddPost />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Head
          description="Aqui, você poderá criar e gerenciar facilmente tópicos para compartilhar com o comunidade! Incluímos alguns dos tópicos mais usados, para que você possa se divertir com eles!"
          img="/img/group-bottom.png"
          project="Udacity - Leitura"
/>,
        <withRouter(Connect(AddPostContainer)) />,
        <Footer
          project="Udacity - Leitura"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "description": "Aqui, você poderá criar e gerenciar facilmente tópicos para compartilhar com o comunidade! Incluímos alguns dos tópicos mais usados, para que você possa se divertir com eles!",
          "img": "/img/group-bottom.png",
          "project": "Udacity - Leitura",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "project": "Udacity - Leitura",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Head
            description="Aqui, você poderá criar e gerenciar facilmente tópicos para compartilhar com o comunidade! Incluímos alguns dos tópicos mais usados, para que você possa se divertir com eles!"
            img="/img/group-bottom.png"
            project="Udacity - Leitura"
/>,
          <withRouter(Connect(AddPostContainer)) />,
          <Footer
            project="Udacity - Leitura"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "description": "Aqui, você poderá criar e gerenciar facilmente tópicos para compartilhar com o comunidade! Incluímos alguns dos tópicos mais usados, para que você possa se divertir com eles!",
            "img": "/img/group-bottom.png",
            "project": "Udacity - Leitura",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "project": "Udacity - Leitura",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
